core/io/resource.h

```diff
@@ -72,6 +72,8 @@ class Resource : public RefCounted {
 	String import_path;
 #endif
 
+	bool freeze_emit_changed = false;
+	bool emit_changed_after_unfreeze = false;
```

Comment:
```suggestion
    enum EmitChangedState {
        EMIT_CHANGED_UNBLOCKED,
        EMIT_CHANGED_BLOCKED,
        EMIT_CHANGED_UNBLOCKED_PENDING_EMIT,
    };
```

Reasons:
1. Use a more concice single variable with all the needed states, and none else.
2. Use the term "block," that is already used in Godot in similar contexts (i.e., "signals blocked").

Reply:
Changed to single variable and "block".